dnf-behave-tests/features/builddep.feature

```diff
@@ -80,3 +80,63 @@ Scenario: Builddep on SPEC with non-available Source0
 
    Error: Some packages could not be found.
    """
+
+@bz1758459
+Scenario: I exclude the highest verion of a package and call dnf builddep with --best
+  Given I use repository "dnf-ci-fedora-updates"
+    And I enable plugin "builddep"
+  Given I create file "dummy-pkg.spec" with
+   """
+   Name: dummy-pkg
+   Summary: dummy-pkg summary
+   Version: 1.0
+   Release: 1
+   License: GPL
+   BuildRequires: flac
+   %description
+   This is a dummy-pkg description
+   %build
+   %files
+   %changelog
+   """
+   When I execute dnf with args "builddep {context.dnf.installroot}/dummy-pkg.spec --exclude flac-1.3.3-3.fc29 --best"
+   Then the exit code is 0
+    And Transaction is following
+        | Action                | Package                    |
+        | install               | flac-0:1.3.3-2.fc29.x86_64 |
+
+@bz1758459
+Scenario: I call dnf builddep with --best on a spec file with a modular dependency (tests handling modular excludes)
+  Given I use repository "dnf-ci-fedora-modular"
+    And I use repository "dnf-ci-fedora"
+    And I enable plugin "builddep"
+    And I execute dnf with args "module enable nodejs:8"
```

Comment:
Why add the module enable? I'm not sure it has any effect on the result, I would guess not, but... The version without the module enable is what @j-mracek reviewed so I'd stick with that. It's shorter too.

Reply:
Without this code I got the error:
`Assertion Failed: [dnf] Following packages weren't captured in the table for action 'module-stream-enable': nodejs:8`

Reply:
Ok, the enabling of the module stream happened automatically in the transaction and you can add it to the transaction like this:
```gherkin
    And Transaction is following
      | Action               | Package                                       |
      ...
      | module-stream-enable | nodejs:8                                      |
```

The way you solved it just took the enable out of the last command output. It's probably functionally equivalent so I'll leave it up to you whether to keep it as is, my way is probably a bit more succinct. The `And modules state is following` at the end is redundant in this test I think...